neo3/wallet/wallet.py

```diff
@@ -0,0 +1,202 @@
+                  "properties": {},
+                  "additionalProperties": True
+                  },
+    },
+    "required": ["path", "name", "scrypt", "accounts", "extra"],
+    "$defs": {
+        "account": {
+            "type": "object",
+            "properties": {
+                "address": {"type": "string"},
+                "label": {"type": "string"},
+                "is_default": {"type": "boolean"},
+                "lock": {"type": "boolean"},
+                "key": {"type": "string"},
+                "contract": {"type": ""},
+                "extra": {"type": ["object", "null"],
+                          "properties": {},
+                          "additionalProperties": True}
+            },
+            "required": ["address", "label", "is_default", "lock", "key", "contract", "extra"]
+
+        },
+        "scrypt_parameters": {
+            "type": "object",
+            "properties": {
+                "n": {"type": "integer"},
+                "r": {"type": "integer"},
+                "p": {"type": "integer"},
+                "length": {"type": "integer"}
+            },
+            "required": ["n", "r", "p"]
+        }
+    }
+}
+
+
+class Wallet(IJson):
+
+    _wallet_version = '3.0'
+
+    def __init__(self,
+                 path: str,
+                 version: str,
+                 scrypt: ScryptParameters,
+                 name: Optional[str] = None,
+                 accounts: List[Account] = None,
+                 extra: Optional[Dict[Any, Any]] = None):
+        """
+        Args:
+            path: the JSON's path
+            name: a label that the user has given to the wallet
+            version: the wallet's version, must be equal or greater then 3.0
+            scrypt: a ScryptParameters object which describes the parameters of the SCrypt algorithm used for encrypting
+                    and decrypting the private keys in the wallet.
+            accounts: an array of Account objects which describe the details of each account in the wallet.
+            extra: an object that is defined by the implementor of the client for storing extra data. This field can be
+                   None.
+        """
+
+        self.path = path
+        self.name = name
+        self.version = version
+        self.scrypt = scrypt
+        self.accounts = accounts if accounts is not None else []
+        self.extra = extra
+
+    @classmethod
+    def default(cls, path: str, name: Optional[str] = None) -> Wallet:
+        """
+        Create a new Wallet with the default settings.
+
+        Args:
+            path: the JSON's path.
+            name: the Wallet name.
+        """
+        return cls(path=path,
+                   name=name,
+                   version=cls._wallet_version,
+                   scrypt=ScryptParameters.default(),
+                   accounts=[],
+                   extra=None)
+
+    @classmethod
+    def new_wallet(cls, location: str):
```

Comment:
I think we should raise `ValueError`s instead of returning `None`. If I call `Wallet.new_wallet` and get `None` as return value then I do not know if this is because the `location` is incorrectly formatted (missing .json) or because the file already exists. If the file exists we might want to give the user the ability to override the existing file.

Reply:
I see.
I've changed to raise a `ValueError` when `location` format is incorrect and also included an optional argument to allow to overwrite if the file exists